src/wallet/message_store.cpp

```diff
@@ -0,0 +1,1389 @@
+  // down still into "message_transporter" for that is a little bit too much
+  struct options {
+    const command_line::arg_descriptor<std::string> bitmessage_address = {"bitmessage-address", mms::message_store::tr("Use PyBitmessage instance at URL <arg>"), "http://localhost:8442/"};
+    const command_line::arg_descriptor<std::string> bitmessage_login = {"bitmessage-login", mms::message_store::tr("Specify <arg> as username:password for PyBitmessage API"), "username:password"};
+  };
+}
+
+void message_store::init_options(boost::program_options::options_description& desc_params)
+{
+  const options opts{};
+  command_line::add_arg(desc_params, opts.bitmessage_address);
+  command_line::add_arg(desc_params, opts.bitmessage_login);
+}
+
+void message_store::init(const multisig_wallet_state &state, const std::string &own_label,
+                         const std::string &own_transport_address, uint32_t coalition_size, uint32_t threshold)
+{
+  m_coalition_size = coalition_size;
+  m_threshold = threshold;
+  m_members.clear();
+  m_messages.clear();
+  m_next_message_id = 1;
+
+  coalition_member member;
+  member.label.clear();
+  member.transport_address.clear();
+  member.monero_address_known = false;
+  memset(&member.monero_address, 0, sizeof(cryptonote::account_public_address));
+  member.index = 0;
+  member.auto_config_public_key = crypto::null_pkey;
+  member.auto_config_secret_key = crypto::null_skey;
+  member.auto_config_running = false;
+
+  for (size_t i = 0; i < m_coalition_size; ++i)
+  {
+    member.me = member.index == 0;    // Simple convention/automatism for now: The very first member is fixed as / must be "me"
+    m_members.push_back(member);
+    member.index++;
+  }
+
+  set_member(state, 0, own_label, own_transport_address, state.address);
+
+  m_nettype = state.nettype;
+  set_active(true);
+  m_filename = state.mms_file;
+  save(state);
+}
+
+void message_store::set_options(const boost::program_options::variables_map& vm)
+{
+  const options opts{};
+  auto bitmessage_address = command_line::get_arg(vm, opts.bitmessage_address);
+  auto bitmessage_login = command_line::get_arg(vm, opts.bitmessage_login);
+  set_options(bitmessage_address, bitmessage_login);
+}
+
+void message_store::set_options(const std::string &bitmessage_address, const std::string &bitmessage_login)
+{
+  m_transporter.set_options(bitmessage_address, bitmessage_login);
+}
+
+void message_store::set_member(const multisig_wallet_state &state,
+                               uint32_t index,
+                               const boost::optional<std::string> &label,
+                               const boost::optional<std::string> &transport_address,
+                               const boost::optional<cryptonote::account_public_address> monero_address)
+{
+  THROW_WALLET_EXCEPTION_IF(index >= m_coalition_size, tools::error::wallet_internal_error, "Invalid member index " + index);
+  coalition_member &m = m_members[index];
```

Comment:
Would you use the expression `m_members.size()` in all places where I now use `m_coalition_size` (about 7 places) to avoid any possibility of the two values going of sync? (Eliminating of course any need for a member variable `m_coalition_size` then.) If yes, don't know ... that somehow goes against my "programming sense" - I would value the simpler and more expressive `m_coalition_size` references higher than the chance to introduce a bug through that separate variable ...

Reply:
OK, fair enough.